### PR TITLE
TASK: Add types to workspace model

### PR DESCRIPTION
**Pull request by daniellienert**

### CODE REVIEW

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 1]
```diff
@@ -245,7 +245,7 @@ public function getOwner()
      * @param UserInterface|string|null $user The new user, or user's UUID
      * @api
      */
-    public function setOwner($user)
+    public function setOwner($user): void
```
Comment (kdambekalns) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php]:
```suggestion
    public function setOwner(?UserInterface $user): void
```

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 2]
```diff
@@ -245,7 +245,7 @@ public function getOwner()
      * @param UserInterface|string|null $user The new user, or user's UUID
      * @api
      */
-    public function setOwner($user)
+    public function setOwner(?UserInterface $user): void
```
Comment (albe) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php]: I think this is not correct, as the code still handles string $user

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 3]
```diff
@@ -332,7 +332,7 @@ public function setBaseWorkspace(Workspace $baseWorkspace)
      * @return Workspace
```
Comment (albe) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php]:
```suggestion
     * @return Workspace|null
```

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 4]
```diff
@@ -231,7 +231,7 @@ public function setDescription($description)
      * @return UserInterface
```
Comment (albe) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php]:
```suggestion
     * @return UserInterface|null
```

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 5]
```diff
@@ -697,7 +697,7 @@ protected function verifyPublishingTargetWorkspace(Workspace $targetWorkspace)
      * @param Workspace $targetWorkspace The target workspace to look in
      * @return NodeData Either a regular node, a shadow node or null
```
Comment (albe) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php]:
```suggestion
     * @return NodeData|null Either a regular node, a shadow node or null
```

**Comment by Sebobo** [comment]
Is this change safe for 5.3 or meant for 6.0?

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 6]
```diff
@@ -208,9 +208,9 @@ public function setTitle($title)
      * @return string
      * @api
      */
-    public function getDescription()
+    public function getDescription(): string
     {
-        return $this->description;
+        return (string)$this->description;
```
Comment (kdambekalns) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php:213]: The typecast is added to handle `null` in existing data? If so, what about `$title` in `getTitle()`?

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 7]
```diff
@@ -33,7 +33,7 @@ class Workspace
     /**
      * This prefix determines if a given workspace (name) is a user workspace.
      */
-    const PERSONAL_WORKSPACE_PREFIX = 'user-';
+    public const PERSONAL_WORKSPACE_PREFIX = 'user-';
```
Comment (kdambekalns) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php:36]: Is it used elsewhere? If not, make it protected (or even private)?

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php [hunk 8]
```diff
@@ -58,7 +58,7 @@ class Workspace
      * @ORM\Column(type="text", length=500, nullable=true)
      * @Flow\Validate(type="StringLength", options={ "minimum"=0, "maximum"=500 })
      */
-    protected $description;
+    protected $description = '';
```
Comment (kdambekalns) [inline Neos.ContentRepository/Classes/Domain/Model/Workspace.php:61]: Remove the `nullable=true` and create a migration, to pull through. 😎 Or leave it as nullable in the code here and below…

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php (hunk 6 above)
Comment (daniellienert) [reply Neos.ContentRepository/Classes/Domain/Model/Workspace.php:213]: Title was not configured to be nullable. So I assumed it cannot be null.

File: Neos.ContentRepository/Classes/Domain/Model/Workspace.php (hunk 7 above)
Comment (daniellienert) [reply Neos.ContentRepository/Classes/Domain/Model/Workspace.php:36]: It is, in `Neos\ContentRepository\Domain\Model\Workspace\User`

**Comment by daniellienert** [comment]
@kdambekalns, @albe thanks for the reviews. Would you have another look?

**Comment by daniellienert** [comment]
@Sebobo they should be pretty safe for 5.3

**Pull request merged**